pkg/controller/component/available.go

```diff
@@ -283,7 +271,7 @@ func (h *AvailableEventHandler) getCachedEvents(comp *appsv1.Component) ([]probe
 }
 
 func (h *AvailableEventHandler) updateCachedEvents(comp *appsv1.Component, events []probeEvent) error {
-	if comp.Annotations == nil && len(events) == 0 {
+	if comp.Status.Message == nil && len(events) == 0 {
```

Comment:
unnecessary to check `comp.Status.Message == nil`?

Reply:
It needs to delete expired events.